src/variables/utils/getWindowVars.ts

```diff
@@ -227,3 +231,83 @@ export const getVariableForZoomRequery = (
       return variable
   }
 }
+
+/*
+ * Prevents the windowPeriod from being null for the time axis
+ *   this fallback is used only for the zoom re-query feature
+ *   if the windowPeriod cannot be found, apply the user-selected domain
+ *     to the original data set and divide by the optimal number of graph points
+ *     to get the estimated windowPeriod
+ */
+export const normalizeWindowPeriodForZoomRequery = (
+  windowPeriod: number | null,
+  timeRange: TimeRange,
+  domain: Array<number>,
+  column: NumericColumnData | string[]
+): number => {
+  if (windowPeriod || !timeRange || !column) {
+    return windowPeriod
+  }
+
+  if (column.length === 0 || domain.length !== 2) {
+    return FALLBACK_WINDOW_PERIOD
+  }
+
+  let counter = 0
+  let startIndex = counter
+  let endIndex = counter
+
+  while (counter < column.length) {
+    if (domain[0] > column[counter]) {
+      startIndex = counter
+    }
+    if (domain[1] > column[counter]) {
+      endIndex = counter
+    }
+    counter += 1
+  }
+
+  const duration =
+    (Number(column[endIndex]) - Number(column[startIndex])) /
+    DESIRED_POINTS_PER_GRAPH
+
+  if (duration !== duration) {
```

Comment:
is this to guard against `NaN`? If so, could [`isNaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN) be used instead?

Reply:
I always get confused whether `isNaN` or `Number.isNaN` is better and usually end up with this equality check as the quicker way of guarding against `NaN`.

But you are right, we should use a built-in checker for this.

The correct one is... `Number.isNaN` so I will go with that.